File: drift/src/main/java/drift/com/drift/activities/DriftActivity.java
package drift.com.drift.activities;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;
import android.net.ConnectivityManager;
import android.support.annotation.Nullable;
import android.support.v4.content.LocalBroadcastManager;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import drift.com.drift.helpers.NetworkReciever;
import drift.com.drift.helpers.Notification;

public class DriftActivity extends AppCompatActivity {

    private NetworkReciever networkChangeReciever = new NetworkReciever();

    private BroadcastReceiver networkChanged;


    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        networkChanged = new BroadcastReceiver() {
            @Override
            public void onReceive(Context context, Intent intent) {
                networkDidUpdate();
            }
        };
    }

    @Override
    protected void onStart() {
        super.onStart();

        IntentFilter filter = new IntentFilter(ConnectivityManager.CONNECTIVITY_ACTION);
        registerReceiver(networkChangeReciever, filter);
        LocalBroadcastManager.getInstance(this).registerReceiver(networkChanged, new IntentFilter(Notification.NETWORK_STATE_CHANGED));
    }

    @Override
    protected void onStop() {
        super.onStop();
        unregisterReceiver(networkChangeReciever);
        LocalBroadcastManager.getInstance(this).unregisterReceiver(networkChanged);
    }


    @Override
    protected void onResume() {
        super.onResume();
        networkChangeReciever.updateNetowrkState(this);
    }

    private void networkDidUpdate(){
        if (NetworkReciever.isInternetConnected(this)) {
            refreshData();
        } else {
            networkNotAvailable();
        }
    }

    public void refreshData() {

    }

    public void networkNotAvailable() {

    }
}

File: drift/src/main/java/drift/com/drift/activities/DriftActivity.kt
package drift.com.drift.activities

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.content.IntentFilter
import android.net.ConnectivityManager
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle

import androidx.localbroadcastmanager.content.LocalBroadcastManager
import drift.com.drift.helpers.NetworkReciever
import drift.com.drift.helpers.Notification

internal open class DriftActivity : AppCompatActivity() {

    private val networkChangeReciever = NetworkReciever()

    private var networkChanged: BroadcastReceiver? = null


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        networkChanged = object : BroadcastReceiver() {
            override fun onReceive(context: Context, intent: Intent) {
                networkDidUpdate()
            }
        }
    }

    override fun onStart() {
        super.onStart()

        val filter = IntentFilter(ConnectivityManager.CONNECTIVITY_ACTION)
        registerReceiver(networkChangeReciever, filter)
        LocalBroadcastManager.getInstance(this).registerReceiver(networkChanged!!, IntentFilter(Notification.NETWORK_STATE_CHANGED))
    }

    override fun onStop() {
        super.onStop()
        unregisterReceiver(networkChangeReciever)
        LocalBroadcastManager.getInstance(this).unregisterReceiver(networkChanged!!)
    }


    override fun onResume() {
        super.onResume()
        networkChangeReciever.updateNetworkState(this)
    }

    private fun networkDidUpdate() {
        if (NetworkReciever.isInternetConnected(this)) {
            refreshData()
        } else {
            networkNotAvailable()
        }
    }

    open fun refreshData() {

    }

    open fun networkNotAvailable() {

    }
}
